src/main/java/com/redhat/foreman/ForemanSharedNodeCloud.java

```diff
@@ -152,9 +152,8 @@ public boolean canProvision(Label label) {
         Map<String, String> hostsMap = getForemanAPI().getCompatibleHosts();
         Set<String> hosts = hostsMap.keySet();
         for (String host: hosts) {
-            boolean match = label.matches(Label.parse(hostsMap.get(host)));
-            if (match) {
-                    return true;
+            if (label == null || label.matches(Label.parse(hostsMap.get(host)))) {
```

Comment:
I prefer yes (or add an option to the cfg. if we don't like this behavior out of the box/by default). 
The idea behind is that we can return true only if there is an available node. So we can't move the check for null on the top of method because we don't know there if we have a spare node.

If we return true (when label == null) in canProvision(), we have to deal with it on all places below.

Anyway - JENKINS-38366 is related to this too.